src/OrbitGl/TimeGraph.cpp

```diff
@@ -724,6 +880,30 @@ void TimeGraph::DrawAllElements(PrimitiveAssembler& primitive_assembler,
   }
 }
 
+void TimeGraph::DoDraw(orbit_gl::PrimitiveAssembler& primitive_assembler,
+                       orbit_gl::TextRenderer& /*text_renderer*/,
+                       const DrawContext& /*draw_context*/) {
+  ORBIT_SCOPE("TimeGraph::DoDraw");
+
+  // TODO(b/230442062): Move IsThisElementPicked directly to Slider methods.
+  if (GetCaptureTimeSpanNs() > 0) {
+    slider_->Draw(primitive_assembler,
+                  primitive_assembler.GetPickingManager()->IsThisElementPicked(slider_.get()));
+    if (vertical_slider_->IsVisible()) {
+      vertical_slider_->Draw(
+          primitive_assembler,
+          primitive_assembler.GetPickingManager()->IsThisElementPicked(vertical_slider_.get()));
+    }
+  }
```

Comment:
You should never call `Draw` on children explicitly. Instead, they should be included in the vector returned in `GetAllChildren()`, and everything around drawing and updating will be taken care of by `CaptureViewElement`. But for this to work, you can't use your own `Draw` overloads.

Since you've already added the sliders to `GetAllChildren()`, they are probably rendered twice currently.

Reply:
You can easily model the `if` conditions then by overriding `GlSlider::ShouldBeRendered()`:

```
bool GlSlider::ShouldBeRenderer() {
   return CaptureViewElement::ShouldBeRendered() && (timeline_info_->GetCaptureTimeSpanNs() > 0);
}
```

`CaptureViewElement` will only render children that return `ShouldBeRendered(): true`.

Reply:
Yes, you are right. We are still rendering it only once, because DoDraw is still not doing anything. I forgot about this part during the refactor, since initially I was going to split it in a few PRs. Thanks!

Reply:
We don't need this for the sliders anymore since they will be drawn as soon as TimeGraph is created.